src/analysis/calculus/specific_functions.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2020 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import analysis.calculus.extend_deriv analysis.calculus.iterated_deriv analysis.complex.exponential
+
+/-!
+# Smoothness of specific functions
+
+The real function `exp_neg_inv_glue` given by `x ↦ exp (-1/x)` for `x > 0` and `0`
+for `x ≤ 0` is a basic building block to construct smooth partitions of unity. We prove that it
+is `C^∞` in `exp_neg_inv_glue_smooth`.
+-/
+
+noncomputable theory
+open_locale classical topological_space
+
+namespace partition_of_unity
+open polynomial real filter set
+
+/-- `exp_neg_inv_glue` is the real function given by `x ↦ exp (-1/x)` for `x > 0` and `0`
+for `x ≤ 0`. is a basic building block to construct smooth partitions of unity. Its main property
+is that it vanishes for `x ≤ 0`, it is positive for `x > 0`, and the junction between the two
+behaviors is flat enough to retain smoothness. The fact that this function is `C^∞` is proved in
+`exp_neg_inv_glue_smooth`. -/
+def exp_neg_inv_glue (x : ℝ) : ℝ := if x ≤ 0 then 0 else exp (-x⁻¹)
+
+/-- Our goal is to prove that `exp_neg_inv_glue` is `C^∞`. For this, we compute its successive
+derivatives for `x > 0`. The `n`-th derivative is of the form `P_n(x) exp(-1/x) / x^(2 n)`,
+where `P_n` is computed inductively. We define `exp_neg_inv_aux_poly n` to be this `P_n`. -/
+noncomputable def exp_neg_inv_aux_poly : ℕ → polynomial ℝ
```

Comment:
Does it make sense to create a local notation for this polynomial? It's very specific to this file, and I think it will make proofs more readable, because now the name is quite long.

Reply:
You could consider putting everything that follows in the `exp_neg_inv_glue` namespace, that would lead to names that are locally in this file short and understandable, and if by chance someone uses them outside this file, then it will most likely be with the namespace prefixed in front of it.

Reply:
Done with a namespace. I used `P_aux` for the auxiliary polynomial and `f_aux` for the auxiliary derivative, and indeed this is much more readable now.